include/FreeRTOS_ND.h

```diff
@@ -0,0 +1,162 @@
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+#ifndef FREERTOS_ND_H
+    #define FREERTOS_ND_H
+
+    #ifdef __cplusplus
+        extern "C" {
+    #endif
+
+/* Application level configuration options. */
+    #include "FreeRTOSIPConfig.h"
+    #include "FreeRTOSIPConfigDefaults.h"
+    #include "IPTraceMacroDefaults.h"
+
+    #if ( ipconfigUSE_IPv6 != 0 )
+/*-----------------------------------------------------------*/
+/* Miscellaneous structure and definitions. */
+/*-----------------------------------------------------------*/
+
+        typedef struct xND_CACHE_TABLE_ROW
+        {
+            IPv6_Address_t xIPAddress; /* The IP address of an ND cache entry. */
+            MACAddress_t xMACAddress;  /* The MAC address of an ND cache entry. */
+            struct xNetworkEndPoint * pxEndPoint;
+            uint8_t ucAge;             /* A value that is periodically decremented but can also be refreshed by active communication.  The ND cache entry is removed if the value reaches zero. */
+            uint8_t ucValid;           /* pdTRUE: xMACAddress is valid, pdFALSE: waiting for ND reply */
+        } NDCacheRow_t;
+
+/*
+ * If ulIPAddress is already in the ND cache table then reset the age of the
+ * entry back to its maximum value.  If ulIPAddress is not already in the ND
+ * cache table then add it - replacing the oldest current entry if there is not
+ * a free space available.
+ */
+        void vNDRefreshCacheEntry( const MACAddress_t * pxMACAddress,
+                                   const IPv6_Address_t * pxIPAddress,
+                                   NetworkEndPoint_t * pxEndPoint );
+
+        #if ( ipconfigUSE_ARP_REMOVE_ENTRY != 0 )
+
+/*
+ * In some rare cases, it might be useful to remove a ND cache entry of a
+ * known MAC address to make sure it gets refreshed.
+ */
+            uint32_t ulNDRemoveCacheEntryByMac( const MACAddress_t * pxMACAddress );
+
+        #endif /* ipconfigUSE_ARP_REMOVE_ENTRY != 0 */
+
+/* Options that can be send after the ICMPv6 header. */
```

Comment:
```suggestion
/* Options that can be sent after the ICMPv6 header. */
```

Reply:
Changed send -> sent in one of the later commits. Thanks